lib/theme_check/checks/asset_size_css.rb

```diff
@@ -0,0 +1,98 @@
+# frozen_string_literal: true
+module ThemeCheck
+  # Reports errors when trying to use too much JavaScript on page load
+  # Encourages the use of the Import on Interaction pattern [1].
+  # [1]: https://addyosmani.com/blog/import-on-interaction/
+  class AssetSizeCSS < LiquidCheck
+    include RegexHelpers
+    severity :error
+    category :performance
+    doc docs_url(__FILE__)
+
+    Link = Struct.new(:href, :index)
+
+    TAG = /#{Liquid::TagStart}.*?#{Liquid::TagEnd}/om
+    VARIABLE = /#{Liquid::VariableStart}.*?#{Liquid::VariableEnd}/om
+    START_OR_END_QUOTE = /(^['"])|(['"]$)/
+    LINK_TAG_HREF = %r{
+      <link
+        (?=(?:[^>]|\n|\r)+?rel=['"]?stylesheet['"]?)+?  # Make sure rel=stylesheet is in the link with lookahead
+        [^>]+                                           # any non closing tag character
+        href=                                           # href attribute start
+        (?<href>
+          '(?:#{TAG}|#{VARIABLE}|[^']+)*'|              # any combination of tag/variable or non straight quote inside straight quotes
+          "(?:#{TAG}|#{VARIABLE}|[^"]+)*"               # any combination of tag/variable or non double quotes inside double quotes
+        )
+        [^>]*                                           # any non closing character till the end
+      >
+    }omix
+    STYLESHEET_TAG = %r{
+      #{Liquid::VariableStart}          # VariableStart
+      (?:(?!#{Liquid::VariableEnd}).)*? # anything that isn't followed by a VariableEnd
+      \|\s*asset_url\s*                 # | asset_url
+      \|\s*stylesheet_tag\s*            # | stylesheet_tag
+      #{Liquid::VariableEnd}            # VariableEnd
+    }omix
+
+    attr_reader :threshold_in_bytes
+
+    def initialize(threshold_in_bytes: 100_000)
+      @threshold_in_bytes = threshold_in_bytes
+    end
+
+    def on_document(node)
+      @node = node
+      @source = node.template.source
+      record_offenses
+    end
+
+    def record_offenses
+      stylesheets(@source).each do |stylesheet|
+        file_size = href_to_file_size(stylesheet.href)
+        next if file_size.nil?
+        next if file_size <= threshold_in_bytes
+        add_offense(
+          "CSS on every page load exceding compressed size threshold (#{threshold_in_bytes} Bytes).",
+          node: @node,
```

Comment:
If this offense is template-wide, you should pass `template: node.template` instead.

Reply:
It isn't. It highlights the URL (or liquid variable) of imported CSS files that are too big.

Reply:
But since you using the `@node` from the `on_document`, the offense will always point the whole document.

That's the top level node of the Liquid AST.